services/backend/src/recommendation_system/recommendation_flow/model_prediction/AlphaModel.py

```diff
@@ -56,9 +58,14 @@ def predict_probabilities(self, content_ids, user_id, seed=None, **kwargs):
 
             # attach embed matrix
             train_content_ids = df.content_id.tolist()
+
+            inv_map = {v: k for k, v in INDEX_TO_CONTENT_ID.items()}
+
             embed_matrix = []
             for content_id in train_content_ids:
-                embed_matrix.append(dic_id_embed[content_id])
+                my_index = inv_map[content_id]
```

Comment:
why do you need this inverse map? id_to_embedding is already content id to embedding

Reply:
Thank you for your suggestion. I have to use that map from content_id to index and then to embedding since the file id_to_embedding.pkl provides a list variable instead of a dictionary, so it will take a lot of time to iterate through the list and get the exact element containing the correct embeddings. On the other hand, we need to get the index to immediately find the item containing the embeddings of that content_id. Likewise, the correlation between the index and the content_id is not consistent: 
while id_embed[10000] includes the embedding of content_id 38598, id_embed[107000] includes that of content_id=135600. Hence, we could not directly calculate the index by adding or subtracting the content_id. In addition, since the function ANN is not run in the f function creating the application, the dictionary CONTENT_ID_TO_INDEX is not global. Thus, I have to inverse the index_to_content_id dictionary to get the index from content_id and then find the embeddings of that content_id by the index.

Reply:
but they are different indexes, just create a new map then in memory to make your life simple

Reply:
Thank you. I will close this pull request and try what you just suggested.